### DISCUSSION TITLE
yaml module on nightly Catalina crashes

### DISCUSSION THREAD

**Discussion by paaguti** [orbitalquark/textadept#119] [General]
Hi,

macOS Catalina is a PITA when it comes to using external libraries, etc. 
In my case, I need to recompile any module that binds to a library. This is the case in the yaml mode.
So what I do is
```
make clean
make deps
make CROSS_OSX=gcc ta=/Applications/Textadept.app/Contents/Resources yamlosx.so
```
This works on the latest beta. 

On the nightly, I can compile but then get a message:
```
error loading module 'yaml.yamlosx' from file '/Users/paag/.textadept/modules/yaml/yamlosx.so':
	dlopen(/Users/paag/.textadept/modules/yaml/yamlosx.so, 6): Symbol not found: _lua_newuserdata
  Referenced from: /Users/paag/.textadept/modules/yaml/yamlosx.so
  Expected in: flat namespace
 in /Users/paag/.textadept/modules/yaml/yamlosx.so
```
when I try to use the module

**Comment by orbitalquark** [comment]
Lua 5.4, which is what Textadept uses now, does not define the symbol `lua_newuserdata`. Instead, it has `lua_newuserdatauv` and a macro define of `lua_userdatauv`. The `ta=/path/to/textadept` should point to Textadept's source that has *src/lua* source. I suspect your */Applications/Textadept.app/...* does not have Lua source inside it. Do you have Lua 5.1-5.3 installed locally on your system? It's possible that the build process is picking that up instead.

I downloaded the nightly module zip and searched inside *yamlosx.so* and only found references to `lua_newuserdatauv`.

**Comment by paaguti** [comment]
Let me check... I'm on another computer now

On Fri, 9 Jul 2021 at 14:05, orbitalquark ***@***.***> wrote:

> Lua 5.4, which is what Textadept uses now, does not define the symbol
> `lua_newuserdata`. Instead, it has `lua_newuserdatauv` and a macro define
> of `lua_userdatauv`. The `ta=/path/to/textadept` should point to
> Textadept's source that has *src/lua* source. I suspect your
> */Applications/Textadept.app/...* does not have Lua source inside it. Do
> you have Lua 5.1-5.3 installed locally on your system? It's possible that
> the build process is picking that up instead.
>
> I downloaded the nightly module zip and searched inside *yamlosx.so* and
> only found references to `lua_newuserdatauv`.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/119#discussioncomment-983906>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4GINWIH2IH47NBSUWTTW3QX7ANCNFSM5ACHAUCA>
> .
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

**Comment by paaguti** [comment]
But anyhow. we have to find a way for this Apple control mania. This will
happen to any extension that uses a local .so

On Fri, 9 Jul 2021 at 16:40, Pedro Andres Aranda Gutierrez <
***@***.***> wrote:

> Let me check... I'm on another computer now
>
> On Fri, 9 Jul 2021 at 14:05, orbitalquark ***@***.***>
> wrote:
>
>> Lua 5.4, which is what Textadept uses now, does not define the symbol
>> `lua_newuserdata`. Instead, it has `lua_newuserdatauv` and a macro define
>> of `lua_userdatauv`. The `ta=/path/to/textadept` should point to
>> Textadept's source that has *src/lua* source. I suspect your
>> */Applications/Textadept.app/...* does not have Lua source inside it. Do
>> you have Lua 5.1-5.3 installed locally on your system? It's possible that
>> the build process is picking that up instead.
>>
>> I downloaded the nightly module zip and searched inside *yamlosx.so* and
>> only found references to `lua_newuserdatauv`.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/orbitalquark/textadept/discussions/119#discussioncomment-983906>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAZKU4GINWIH2IH47NBSUWTTW3QX7ANCNFSM5ACHAUCA>
>> .
>>
>
>
> --
> Fragen sind nicht da um beantwortet zu werden,
> Fragen sind da um gestellt zu werden
> Georg Kreisler
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

**Comment by orbitalquark** [comment]
I'm not sure how. I don't have a Mac with Catalina on it for testing :(

**Comment by paaguti** [comment]
Would including the required headers in the Textadept.app help somehow? Just wondering whether that would help us compile the modules correctly?

Enviado desde mi iPhone

**Comment by orbitalquark** [comment]
I'm not sure what that means. If you mean have *lua.h* and friends, then you could try running `make lua` within *Textadept.app/Contents/Resources/src* manually.

**Comment by paaguti** [comment]
I'll try that :-)

On Sun, 11 Jul 2021 at 15:09, orbitalquark ***@***.***> wrote:

> I'm not sure what that means. If you mean have *lua.h* and friends, then
> you could try running `make lua` within
> *Textadept.app/Contents/Resources/src* manually.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/119#discussioncomment-990268>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4CY67CZSIY5K2S62JTTXGJZZANCNFSM5ACHAUCA>
> .
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

**Comment by paaguti** [comment]
That seems the way to go… worked for me (tm) ;-)

Enviado desde mi iPhone
